### DISCUSSION TITLE
Support for Matrix Tests / pytest-like parametrized fixtures

### DISCUSSION THREAD

**Discussion by defreng** [behave/behave#1131] [Ideas]
Hey guys!

First of all, thanks for creating & maintaining behave - it's really an amazing piece of software ❤️ 

I've seen in the docs, that behave has some nice support for fixtures, similar to what pytest offers. Now, there is another really awesome feature in pytest with these fixtures that makes it very easy to test scenarios in a matrix-like pattern with many different combinations:

https://docs.pytest.org/en/7.3.x/how-to/fixtures.html#parametrizing-fixtures

I was wondering how difficult it would be to add support for a similar thing in behave? Do you maybe even have some pointers to code where I could start contributing?

Best,
Alex

**Comment by jenisys** [comment]
`ScenarioOutline`(s) / `ScenarioTemplate`(s) are the corresponding functionality to `pytest parametrized-fixture` (for one dimension).

Multi-dimensional `@parametrize` will require special preprocessing functionality in Gherkin for ScenarioOutline Examples table(s) generation/transformation with `@fixture-tags` (hint: Fixture will need to modify/generate/transform the existing Examples table before `behave` starts processing/selecting which features/scenarios to run). This is currently not supported.

SEE ALSO:
* [docs: Scenario Outline](https://behave.readthedocs.io/en/latest/gherkin/?highlight=Scenario%20Outline#scenario-outlines)
* [docs: Gherkin v6 — Scenario Outline Alias](https://behave.readthedocs.io/en/latest/gherkin/?highlight=Scenario%20Outline#scenario-outlines)
* [features/scenario_outline.basics.feature](https://github.com/behave/behave/blob/main/features/scenario_outline.basics.feature)
* [features/scenario_outline.parametrized.feature](https://github.com/behave/behave/blob/main/features/scenario_outline.parametrized.feature)

RELATED: Table generation/transform on ScenarionOutline Examples
* #464 
* #622 
* #707
